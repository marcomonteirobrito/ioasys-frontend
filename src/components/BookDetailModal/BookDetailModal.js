import React from 'react';
import PropTypes from 'prop-types';

import Modal from '../Modal/Modal';

import {
  Container,
  ImageBookContainer,
  DetailBookContainer,
  BookTitle,
  BookAuthors,
  BookInformationsContainer,
  BookInformations,
  InformationsSpan,
  Title,
  Result,
  DescriptionContainer,
  Description,
  QuotesIcon,
  SubTitle,
} from './styles';

/**
 * @param {boolean} visible - Informa se o modal deve estar visível
 * @param {function} onClose - Função a ser chamada quando clicar no botão de fechar o modal
 */

const BookDetailModal = ({ visible, onClose, bookData }) => (
  <Modal visible={visible} onClose={onClose}>
    <Container>
      <ImageBookContainer>{bookData.imageUrl}</ImageBookContainer>
      <DetailBookContainer>
        <BookTitle>{bookData.title}</BookTitle>

        {bookData.authors.map((author) => (
          <BookAuthors>{author}</BookAuthors>
        ))}
        <BookInformationsContainer>
          <SubTitle>INFORMAÇÕES</SubTitle>
          <BookInformations>
            <InformationsSpan>
              <Title>Páginas</Title>
              <Result>{bookData.pageCount} páginas</Result>
            </InformationsSpan>

            <InformationsSpan>
              <Title>Editora</Title>
              <Result>{bookData.publisher}</Result>
            </InformationsSpan>

            <InformationsSpan>
              <Title>Publicação</Title>
              <Result>{bookData.published}</Result>
            </InformationsSpan>

            <InformationsSpan>
              <Title>Idioma</Title>
              <Result>{bookData.language}</Result>
            </InformationsSpan>

            <InformationsSpan>
              <Title>Título Original</Title>
              <Result>{bookData.title}</Result>
            </InformationsSpan>

            <InformationsSpan>
              <Title>ISBN-10</Title>
              <Result>{bookData.isbn10}</Result>
            </InformationsSpan>

            <InformationsSpan>
              <Title>ISBN-13</Title>
              <Result>{bookData.isbn13}</Result>
            </InformationsSpan>
          </BookInformations>
        </BookInformationsContainer>
        <SubTitle>RESENHA DA EDITORA</SubTitle>
        <DescriptionContainer>
          <QuotesIcon />
          <Description>{bookData.description}</Description>
        </DescriptionContainer>
      </DetailBookContainer>
    </Container>
  </Modal>
);

BookDetailModal.propTypes = {
  visible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  bookData: PropTypes.objectOf(PropTypes.object).isRequired,
};

export default BookDetailModal;
