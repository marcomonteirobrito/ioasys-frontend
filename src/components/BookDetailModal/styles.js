import styled from 'styled-components';
import { ImQuotesLeft } from 'react-icons/im';

import Colors from '../../commons/Colors';

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 769px;
  height: 608px;
`;
export const ImageBookContainer = styled.div`
  width: 349px;
  height: 512px;
  margin-left: 32px;
  margin-right: 16px;
`;
export const DetailBookContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 349px;
  height: 512px;
  margin-right: 32px;
`;
export const BookTitle = styled.h1`
  size: 28px;
  line-height: 40px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 349px;
`;
export const BookAuthors = styled.span`
  color: ${Colors.authorBookDetail};
  size: 12px;
  line-height: 20px;
  margin-top: 4px;
`;

export const SubTitle = styled.h5``;

export const BookInformationsContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin: 32px 0px 20px 0px;
`;

export const BookInformations = styled.div`
  display: flex;
  flex-direction: column;
`;
export const InformationsSpan = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
export const Title = styled.div`
  font-size: 12px;
  line-height: 28px;
`;
export const Result = styled.div`
  color: ${Colors.detailBook};
  font-size: 12px;
  line-height: 28px;
`;
export const DescriptionContainer = styled.div`
  display: flex;
  margin-top: 8px;
`;
export const Description = styled.div`
  width: 330px;
  margin-left: 4px;
  color: ${Colors.detailBook};
`;
export const QuotesIcon = styled(ImQuotesLeft)``;
