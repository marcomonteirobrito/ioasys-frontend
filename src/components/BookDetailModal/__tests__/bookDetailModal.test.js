/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';

import BookDetailModal from '../BookDetailModal';

const bookDataMock = {
  authors: ['Raul Pereira', 'Júlia Pereira Filho', 'Natália Barros Jr.'],
  title: 'Assumenda',
  description:
    'Est cupiditate rerum sed nisi quam dolorem vel qui veniam. Veritatis quam expedita. Dolor unde animi est dolor unde qui nisi. Iusto in saepe repellendus officia omnis aut reiciendis sapiente. Voluptate et at ut. Velit est iure repellat doloribus omnis earum laborum.\n \rModi sapiente laudantium similique recusandae modi eveniet minima sint in. Eum autem nihil est enim a nostrum quaerat magnam. Et tempora iure quidem aliquid rerum accusamus id.',
  pageCount: 990,
  category: 'Biografias',
  imageUrl: 'https://files-books.ioasys.com.br/Book-0.jpg',
  language: 'Inglês',
  isbn10: '6163701883',
  isbn13: '951-6163701883',
  publisher: 'Barros - Xavier',
  published: 2003,
  id: '60171639faf5de22b804a06e',
};

const renderBookDetailModal = () => {
  const onButtonClick = jest.fn();

  return render(
    <BookDetailModal
      onClose={onButtonClick}
      visible={false}
      bookData={bookDataMock}
    >
      Entrar
    </BookDetailModal>
  );
};

describe('BookDetailModal', () => {
  it('Deve renderizar o componente corretamente', () => {
    renderBookDetailModal();

    expect(document.body).toMatchSnapshot();
  });
});
