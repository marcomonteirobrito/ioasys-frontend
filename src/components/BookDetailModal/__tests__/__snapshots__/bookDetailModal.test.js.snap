// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetailModal Deve renderizar o componente corretamente 1`] = `
<body>
  <div />
</body>
`;
